src/ts/components/fullscreentogglebutton.ts

```diff
@@ -20,28 +20,32 @@ export class FullscreenToggleButton extends ToggleButton<ToggleButtonConfig> {
   configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
-    let fullscreenStateHandler = () => {
-      if (player.getViewMode() === player.exports.ViewMode.Fullscreen) {
-        this.on();
-      } else {
-        this.off();
-      }
+    let isFullScreenAvailable = () => {
+      return player.isViewModeAvailable(player.exports.ViewMode.Fullscreen);
     };
 
-    let fullscreenAvailableHandler = () => {
-      if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
-        this.show();
-      } else {
-        this.hide();
-      }
+    let fullscreenStateHandler = () => {
+      player.getViewMode() === player.exports.ViewMode.Fullscreen ? this.on() : this.off();
     };
 
-    uimanager.getConfig().events.onUpdated.subscribe(fullscreenAvailableHandler);
+    let fullscreenAvailabilityChangedHandler = () => {
+      isFullScreenAvailable() ? this.show() : this.hide();
+    };
 
     player.on(player.exports.PlayerEvent.ViewModeChanged, fullscreenStateHandler);
 
+    // Available only in our native SDKs for now
+    if ((player.exports.PlayerEvent as any).ViewModeAvailabilityChanged) {
+      player.on(
+        (player.exports.PlayerEvent as any).ViewModeAvailabilityChanged,
+        fullscreenAvailabilityChangedHandler,
+      );
+    }
+
+    uimanager.getConfig().events.onUpdated.subscribe(fullscreenAvailabilityChangedHandler);
```

Comment:
Regarding the subscribes, should we unsubscribe them here? 
or this is done somewhere else?

Reply:
They get cleared in `UIManager.clearEventHandlers` when the UI itself gets removed. Otherwise, I don't think there is need to unsubscribe 🤔